script.js

```diff
@@ -1,4 +1,146 @@
+// -- SPS Part 1 -- //
+// Users input "scissors","paper", or "stone"
+// Program outputs whether the user won, the program won or draw
+
+// -- SPS Part 2 -- //
+// Keep track of users win and computer wins
+// keep track of users loss and computer loss
+// keep track of draws for both
+
+var currentGameMode = "waiting for user name";
+var totalNumOfRounds = 0;
+var numOfUserWins = 0;
+var numOfComWins = 0;
+var numOfDraws = 0;
+var numOfUserLosses = 0;
+var numOfComLosses = 0;
+var userName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (currentGameMode == "waiting for user name") {
+    // waiting for user input
+    userName = input;
+
+    // Switch to SPS game after userName input
+
+    // Ask user to choose 'reversed' or 'normal' SPS//
+    currentGameMode = "Scissors Paper Stone Game";
+    myOutputvalue = `Hello ${userName}! Please select 'scissors, paper or stone' OR 'reverse scissors, reverse paper or reverse stone!`;
+  } else if (currentGameMode == "Scissors Paper Stone Game") {
+    myOutputvalue = playSPSGame(userName, input);
+  }
+  return myOutputvalue;
+};
+
+// --Normal and Reverse SPS version-- //
+// Normal: scissors > paper; paper > stone; stone > scissors;
+// Reverse: scissors < paper; paper < stone; stone < scissors;
+// Draw if both choose the same subject
+var playSPSGame = function (userName, input) {
```

Comment:
technically don't need to take in `userName `as a parameter, since it is a global variable